merge.go

```diff
@@ -387,9 +388,18 @@ func (m *mergeBuffer) fill() error {
 				return err
 			}
 		}
+		if n == 0 {
+			emptyRowGroups++
+		}
 		m.buffer[i] = m.buffer[i][:n]
 	}
 	heap.Init(m)
+	// heap.Init sorts m.rows pushing empty row groups at the end of the slice.
+	//
+	// We can call heap.Pop(m) n times where n is emptyRowGroups. But it has the same
+	// effect as decrementing m.len while doing extra work. We are avoiding extra
+	// work by directly decrementing m.len by emptyRowGroups.
+	m.len -= emptyRowGroups
```

Comment:
Nice catch, you are right. The comment is wrong, It seems the test is passing by accident because there are no non empty row groups following the empty one.

I'm on mobile now, I will take another proper look tomorrow.